pkg/query-service/app/integrations/messagingQueues/kafka/translator.go

```diff
@@ -0,0 +1,101 @@
+package kafka
+
+import (
+	"fmt"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+var defaultStepInterval int64 = 60
+
+func BuildQueryRangeParams(messagingQueue *MessagingQueue, queryContext string) (*v3.QueryRangeParamsV3, error) {
+
+	var cq *v3.CompositeQuery
+	if queryContext == "producer" {
+		chq, err := buildProducerClickHouseQuery(messagingQueue)
+		if err != nil {
+			return nil, err
+		}
+		cq, err = buildCompositeQueryProducer(chq)
+
+		if err != nil {
+			return nil, err
+		}
+	} else if queryContext == "consumer" {
+		chq, err := buildConsumerClickHouseQuery(messagingQueue)
+		if err != nil {
+			return nil, err
+		}
+		cq, err = buildCompositeQueryConsumer(chq)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	queryRangeParams := &v3.QueryRangeParamsV3{
+		Start:          messagingQueue.Start,
+		End:            messagingQueue.End,
+		Step:           defaultStepInterval,
+		CompositeQuery: cq,
+		Version:        "v4",
+		FormatForWeb:   true,
+	}
+
+	return queryRangeParams, nil
+}
+
+func buildProducerClickHouseQuery(messagingQueue *MessagingQueue) (*v3.ClickHouseQuery, error) {
+	start := messagingQueue.Start
+	end := messagingQueue.End
+	topic, ok := messagingQueue.Variables["topic"]
+	if !ok {
+		return nil, fmt.Errorf("invalid type for Topic")
+	}
+
+	partition, ok := messagingQueue.Variables["partition"]
+
+	if !ok {
+		return nil, fmt.Errorf("invalid type for Partition")
+	}
+	query := generateProducerSQL(start, end, topic, partition)
```

Comment:
Is this prone to sql injection? I have seen folks use prepared statements for CH query

Reply:
No, it's not prone to SQL injection as the query itself is not exposed in the API endpoint, only a few parameters are exposed as request body parameters.

Also, thinking more, I think all our `existing` APIs are prone to SQL injection since users know the payload contains the composite query, and the code is open. They can inject SQL based composite query IMO.
I have created an issue [here](https://github.com/SigNoz/signoz/issues/5566) for query range APIs.